src/rules/max-line-length/index.js

```diff
@@ -50,7 +50,8 @@ export default function (maxLength, options) {
     }
 
     function checkNewline(match) {
-      let nextNewlineIndex = rootString.indexOf("\n", match.endIndex)
+      const endOfLineIndicator = (rootString.indexOf("\r\n", match.endIndex) !== -1) ? "\r\n" : "\n"
```

Comment:
@davidtheclark Unless I've misunderstood the question, that's what the ternary is for in the line selector. If the line is using is using `\r\n` it will use that as the end-of-line indicator and if not, it will use `\n`.